### DISCUSSION TITLE
Is there a way to re-use queries?

### DISCUSSION THREAD

**Discussion by Riezebos** [evidence-dev/evidence#2459] [Q&A]
As far as I can tell, this hasn't been asked before, but please correct me if I'm wrong.

I have a table with multiple sets of columns that I first unpivot in a query and then show the results of in a stacked 100% chart. It would be nice if I can make a re-usable component that includes the query, where I pass a string that is used in the query, the name of the chart, etc. Is this possible?

- I tried using a loop but I don't think I can create a query within a loop
- I looked into trying it with a custom component but I don't see whether it is possible to write a query in a component
- It could work using a templated page, but I have a bunch of filters that automatically reset when switching page, and I want the user to be able to see both bar charts while only needing to set the filters once

Does anyone have experience with this and can help me? Or should I just suck it up and copy paste the query+component a few times and find/replace the string I need to change?

**Comment by Riezebos** [comment]
Ah I think I can solve it with an Input component, but I'd prefer using a query in a loop component or using a query in a custom component. So if anyone knows whether/how that might be possible I'd still be curious to know!

**Comment by archiewood** [comment]
@ItsMeBrianD has a pattern for this with the Query() function
